src/main/java/seedu/address/storage/JsonTaskListStorage.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.storage;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.commons.util.FileUtil;
+import seedu.address.commons.util.JsonUtil;
+import seedu.address.model.ReadOnlyTaskList;
+
+public class JsonTaskListStorage implements TaskListStorage {
+
+    private static final Logger logger = LogsCenter.getLogger(JsonTaskListStorage.class);
+
+    private Path filePath;
+
+    public JsonTaskListStorage(Path filePath) {
+        this.filePath = filePath;
+    }
+
+    public Path getTaskListFilePath() {
+        return filePath;
+    }
+
+    @Override
+    public Optional<ReadOnlyTaskList> readTaskList() throws DataConversionException, IOException {
+        return readTaskList(filePath);
+    }
+
+    /**
+     * Similar to {@link #readTaskList()}.
+     *
+     * @param filePath location of the data. Cannot be null.
+     * @throws DataConversionException if the file is not in the correct format.
+     */
+    public Optional<ReadOnlyTaskList> readTaskList(Path filePath) throws DataConversionException, IOException {
+        requireNonNull(filePath);
+
+        Optional<JsonSerializableTaskList> jsonTaskList = JsonUtil.readJsonFile(
+                filePath, JsonSerializableTaskList.class);
+        if (!jsonTaskList.isPresent()) {
+            return Optional.empty();
+        }
+
+        try {
+            return Optional.of(jsonTaskList.get().toModelType());
+        } catch (IllegalValueException ive) {
+            logger.info("Illegal values found in " + filePath + ": " + ive.getMessage());
+            throw new DataConversionException(ive);
+        }
```

Comment:
Apologies, would the method still work if your signature wrote throws IllegalValueException instead of IOException to make it clearer

Reply:
The `IllegalValueException` exception is thrown by line 52, by the `JsonSerializableTaskList::toModelType` and caught by the catch block, so technically this method does not throw `IllegalValueException`, that's why in the method signature it's not included.